### PR TITLE
fix(aws-eks): fail to update both logging and access at the same time

### PR DESCRIPTION
**Pull request by pahud**
This PR addresses the following known issues:

1. When updating the cluster endpoint access type only with logging predefined yet unchanged, the cluster-resource-handler updates both the logging and access, which is not allowed and throws the SDK error. This PR fixed this and will update access type only, which is allowed.
2. When updating the cluster endpoint public cidr with exactly the same size of cidr, the `setsEqual` function should return correctly.
3. When updating the cluster endpoint public access from one cidr to multiple cidr with logging predefined yet unchanged, the update should return correctly.
4. Updating both access and logging now throws an error from CDK custom resource.

This PR is just a temporary fix that does not implement multiple operations in the cluster-resource-handler custom resource provider(i.e. update both logging and access). 

Fixes: https://github.com/aws/aws-cdk/issues/21439

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts [hunk 1]
```diff
@@ -136,23 +136,32 @@ export class ClusterResourceHandler extends ResourceHandler {
       return this.updateClusterVersion(this.newProps.version);
     }
 
-    if (updates.updateLogging || updates.updateAccess) {
+    if (updates.updateLogging && updates.updateAccess) {
```
Comment (gazal-k) [inline packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts:139]: the EKS API does not allow both to be updated at the same time, but if we make 2 separate API calls, do we need to set this restriction on the custom resource?

Comment (pahud) [reply packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts:139]: I am not sure if this is a good idea to have a custom resource making multiple API calls. Would love to hear comments from @iliapolo or any maintainers before we move this PR further.

Comment (otaviomacedo) [reply packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts:139]: We have precedents for this already in other custom resources. Some examples involving multiple write calls:

* In S3, for automatic deletion of objects, we call `deleteObjects` multiple times.
* In IAM, for the OIDC provider, we call many IAM methods, such as `updateOpenIDConnectProviderThumbprint`, `addClientIDToOpenIDConnectProvider` and `removeClientIDFromOpenIDConnectProvider`.
* In Logs, for log retention, we call `createLogGroup`, `deleteRetentionPolicy` and `putRetentionPolicy`. 

And, if we also consider read calls, pretty much every custom resource makes a combination of read and write calls.

Comment (pahud) [reply packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts:139]: Given that, I agree we should making multiple API calls in this custom resource but I probably don't have capacity for that in the next few weeks. I will make this PR as is for immediate fix and leaving the multiple API calls implementation for another PR if anyone is interested.

**Comment by pahud** [comment]
Hi @otaviomacedo 

This PR is ready for the first review. I didn't add or modify any integ test but I did added a few additional unit tests. Thanks.

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**